### DISCUSSION TITLE
How to migrate firefly to other machine? (Unsupported cipher or incorrect key length)

### DISCUSSION THREAD

**Discussion by azzamsa** [firefly-iii/firefly-iii#8839] [Firefly III]
Hello,

I've successfully migrated all my self-hosted apps to another machine using the following method, which has worked perfectly fine for all of them, including those with a PostgreSQL database:

1. Stop the docker services on Machine A.
2. Copy all the data from Machine A to Machine B using SCP.
3. Start the docker services on Machine B.

However, I'm encountering an issue with Firefly. After the migration, I'm receiving the following error:

```php
firefly-app-1   | [Tue Apr 30 05:41:32.539763 2024] [php:error] [pid 65] [client 127.0.0.1:43208] PHP Fatal error:  Uncaught RuntimeException: Unsupported cipher or incorrect key length. Supported ciphers are: aes-128-cbc, aes-256-cbc, aes-128-gcm, aes-256-gcm. in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/Encrypter.php
```

Additionally, I'm unable to run any commands with `docker exec -t firefly-app-1 bash`. It seems to be stuck, possibly due to the app status showing as 'starting':

```
root@dc0ee9b75e77:/var/www/html# php --version
^C
```

```
root@Kit:~/firefly# docker compose ps
NAME                COMMAND                   SERVICE             STATUS                PORTS
firefly-app-1       "/usr/local/bin/entr…"    app                 running (unhealthy)   0.0.0.0:8083->8080/tcp, :::8083->8080/tcp
firefly-cron-1      "sh -c 'echo \"0 3 * …"   cron                running
firefly-db-1        "docker-entrypoint.s…"    db                  running               3306/
```

**Comment by JC5** [comment]
Is machine B another architecture?

**Reply by azzamsa** [reply]
I *assume* yes, because I did migrate my Firefly from time to time but with the same Linode servers with no errros. This time I migrate them to Vultr.

Any solution? 🙏🏼

**Reply by JC5** [reply]
Sorry, not a clue :(.

Perhaps use something like mysqldump to export the database and import it again?

**Reply by azzamsa** [reply]
I have another self-hosted app that encrypts all the user `CLI command history`, it is https://github.com/atuinsh/atuin. 
But why it doesn't share the same fate as Firefly when moving to new architecture?
Why does Firefly have something like `Unsupported cipher or incorrect key length. Supported ciphers are` ?

And I don't think I can do `full export` and `full import` using the Firefly UI.
